packages/react-components/src/components/table-output-component.tsx

```diff
@@ -263,15 +411,63 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
     }
 
     private onModelUpdated = async () => {
-        if (this.pendingIndex && this.gridApi) {
-            if (this.gridApi.getSelectedNodes().length === 0) {
-                this.gridApi.getDisplayedRowAtIndex(this.pendingIndex).setSelected(true);
-            }
-            this.pendingIndex = undefined;
-        }
+        this.selectRowsByTimestamp();
+        this.selectRowsByIndex();
         if (this.columnArray.length > 0 && !this.columnsPacked && this.columnApi) {
             this.columnApi.autoSizeAllColumns();
             this.columnsPacked = true;
         }
     };
+
+    private selectRowsByTimestamp(): void {
+        if (this.enableTimestampSelection && this.gridApi) {
+
+            if (!this.timestampCol || this.startTimestamp === this.endTimestamp) {
+                if (this.isIndexChanged) {
+                    const rowNode = this.gridApi.getDisplayedRowAtIndex(this.selectStartIndex);
+                    if (rowNode.id) {
+                        rowNode.setSelected(true);
+                        this.isIndexChanged = false;
+                    }
+                }
+            } else {
+                this.gridApi.forEachNode(rowNode => {
+                    if (rowNode.id) {
+                        if (rowNode.data && this.timestampCol &&
+                            rowNode.data[this.timestampCol] >= Math.min(this.startTimestamp, this.endTimestamp) &&
+                            rowNode.data[this.timestampCol] <= Math.max(this.startTimestamp, this.endTimestamp)) {
+                            rowNode.setSelected(true);
+                            if (this.endTimestamp < this.startTimestamp) {
+                                this.selectEndIndex = Math.min(this.selectEndIndex, rowNode.rowIndex);
+                            } else {
+                                this.selectEndIndex = Math.max(this.selectEndIndex, rowNode.rowIndex);
+                            }
+                        } else {
+                            rowNode.setSelected(false);
+                        }
+                    }
+                });
+
+                if (this.gridApi.getSelectedNodes.length === 0) {
+                    this.selectStartIndex = Number.MAX_VALUE;
```

Comment:
So this line can cause an exception when selecting by index.

Here is a log showing the error:

user starts a new single selection by clicking in time graph

root INFO handleTimeSelectionChange() startTimestamp=4699075186 endTimestamp=4699075186  >>> single selection
root INFO handleTimeSelectionChange fetchTableIndex...  >>> this now blocked in an await, UI continues handling other events

user extends the selection by dragging the mouse to the right

root INFO handleTimeSelectionChange() startTimestamp=4699075186 endTimestamp=4699088315  >>> selection is extended, start time does not change
root INFO handleTimeSelectionChange calling selectRowsByTimestamp()
root INFO selectRowsByTimestamp enableTimestampSelection=true startTimestamp=4699075186 endTimestamp=4699088315 selectStartIndex=1.7976931348623157e+308
root INFO selectRowsByTimestamp selectStartIndex=Number.MAX_VALUE  >>> because selectedNodes.length=0

the above happens a few times as mouse is extending selection with new end time

now the trace-server comes back with the start index, the await returns

root INFO handleTimeSelectionChange fetchTableIndex => returned 1238
root INFO handleTimeSelectionChange gridApi.ensureIndexVisible(1238)  >>> this triggers a fetching of new model lines from trace-server in the background
root INFO handleTimeSelectionChange fetchTableIndex selectStartIndex=1238
root INFO handleTimeSelectionChange calling selectRowsByTimestamp()
root INFO selectRowsByTimestamp enableTimestampSelection=true startTimestamp=4699075186 endTimestamp=4699170844 selectStartIndex=1238  <<< when we fetched start index it was a single selection, not anymore
root INFO selectRowsByTimestamp selectStartIndex=Number.MAX_VALUE  >>> because selectedNodes.length=0

now user starts a new single selection by clicking in time graph

root INFO handleTimeSelectionChange() startTimestamp=4699470950 endTimestamp=4699470950  >>> single selection
root INFO handleTimeSelectionChange fetchTableIndex...  >>> this now blocked in an await, UI continues handling other events

now the trace-server comes back with the lines from start index 1238 and updates the model

root INFO onModelUpdated calling selectRowsByTimestamp()
root INFO selectRowsByTimestamp enableTimestampSelection=true startTimestamp=4699470950 endTimestamp=4699470950 selectStartIndex=1.7976931348623157e+308
root INFO selectRowsByTimestamp getDisplayedRowAtIndex(1.7976931348623157e+308)  <<< *** EXCEPTION IS THROWN ***

Reply:
Thanks for this explanation. I have switched back to looping through the active rows and then selecting the row if there is a match. That way we would never select an index that doesn't exist. I don't see the exception now.

Reply:
The conditions to have an invalid selectStartIndex are still there, except that now they just don't select anything instead of cause an exception. So in the following example, the table ends up with no selection:

root INFO *** handleTimeSelectionChange({"start":1341337214,"end":1341337214})
root INFO vvv fetching table index @4738416805
root INFO *** handleTimeSelectionChange({"start":1341337214,"end":1341610057})
root INFO *** handleTimeSelectionChange({"start":1341337214,"end":1341630876})
root INFO *** handleTimeSelectionChange({"start":1341337214,"end":1342068391})
root INFO ^^^ fetching table index returns 4269
root INFO vvv fetching table lines @4200
root INFO *** handleTimeSelectionChange({"start":1341337214,"end":1341630876})
root INFO *** handleTimeSelectionChange({"start":1341337214,"end":1341610057})
root INFO *** handleTimeSelectionChange({"start":1341337214,"end":1341337214})
root INFO selectStartIndex=1.7976931348623157e+308
root INFO ^^^ fetching table lines returns
root INFO selectStartIndex=1.7976931348623157e+308